systemtest/src/test/java/io/strimzi/systemtest/BaseST.java

```diff
@@ -579,294 +616,24 @@ void verifyLabelsForRoleBindings(String clusterName, String appName) {
-        } catch (InterruptedException | ExecutionException | TimeoutException e) {
-            e.printStackTrace();
-            throw e;
-        }
-    }
-
-    public void receiveMessagesExternal(String namespace, String topicName, int messageCount) throws Exception {
-        receiveMessagesExternal(namespace, topicName, messageCount, "my-group-" + new Random().nextInt(Integer.MAX_VALUE));
-    }
-
-    public void receiveMessagesExternalTls(String namespace, String topicName, int messageCount, String userName, String consumerGroup) throws Exception {
-        try (KafkaClient testClient = new KafkaClient()) {
-            Future consumer = testClient.receiveMessagesTls(topicName, namespace, CLUSTER_NAME, userName, messageCount, "SSL", consumerGroup);
-
-            assertThat("Consumer consumed all messages", consumer.get(1, TimeUnit.MINUTES), is(messageCount));
-        } catch (InterruptedException | ExecutionException | TimeoutException e) {
-            e.printStackTrace();
-            throw e;
-        }
-    }
-
-    public void receiveMessagesExternalTls(String namespace, String topicName, int messageCount, String userName) throws Exception {
-        receiveMessagesExternalTls(namespace, topicName, messageCount, userName, "my-group-" + new Random().nextInt(Integer.MAX_VALUE));
+    protected void verifyCRStatusCondition(Condition condition, String status, String type) {
+        verifyCRStatusCondition(condition, null, null, status, type);
     }
 
-    public void receiveMessagesExternalScramSha(String namespace, String topicName, int messageCount, String userName, String consumerGroup) throws Exception {
-        try (KafkaClient testClient = new KafkaClient()) {
-            Future consumer = testClient.receiveMessagesTls(topicName, namespace, CLUSTER_NAME, userName, messageCount, "SASL_SSL", consumerGroup);
+    protected void verifyCRStatusCondition(Condition condition, String message, String reason, String status, String type) {
+        assertThat(condition.getStatus(), is(status));
+        assertThat(condition.getType(), is(type));
 
-            assertThat("Consumer consumed all messages", consumer.get(1, TimeUnit.MINUTES), is(messageCount));
-        } catch (InterruptedException | ExecutionException | TimeoutException e) {
-            e.printStackTrace();
-            throw e;
+        if (condition.getMessage() != null && condition.getReason() != null) {
+            assertThat(condition.getMessage(), containsString(message));
+            assertThat(condition.getReason(), is(reason));
         }
     }
 
-    public void receiveMessagesExternalScramSha(String namespace, String topicName, int messageCount, String userName) throws Exception {
-        receiveMessagesExternalScramSha(namespace, topicName, messageCount, userName, "my-group-" + new Random().nextInt(Integer.MAX_VALUE));
+    void assertNoCoErrorsLogged(long sinceSeconds) {
```

Comment:
Note that here you're perfectly happy to write a method name with camelcase `Co` not `CO`.